### PR TITLE
Primitives now use color closures when printed on the command line

### LINKED ISSUE #7554: Let the closure color config also cover non-table output
### Related problem

#7141 introduced conditional coloring by assigning a closure to a type's color config.
It currently only supports values that are displayed as cells in a table.
grafik

### Describe the solution you'd like

Also cover single values.

### Describe alternatives you've considered

_No response_

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by webbedspace**
# Description

Closes #7554

image

# User-Facing Changes

See above.

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by webbedspace** [comment]
Hmm… I need to fix the test macros so that they strip ANSI from the stdout result… but some tests actually do need to examine the ANSI codes… hmm…

**Comment by sholderbach** [comment]
> Hmm… I need to fix the test macros so that they strip ANSI from the stdout result… but some tests actually do need to examine the ANSI codes… hmm…

Yeah the `nu!` tests need some work, to both faithfully test and run with a minimal config possible (@rgwood did some profiling on how the `default_env.nu` is slowing us down in simple tests https://github.com/nushell/nushell/issues/6677#issuecomment-1364606091). I still have to rework some of the stuff around line ending handling as there edgecases are impossible to test now (#7543)

I would lean towards not stripping silently in the test evaluation and have a minimal test config without colors for most tests.

**Comment by rgwood** [comment]
It would be nice if the `nu!` macro was more configurable; ideally it would be easy to control whether to strip newlines and ANSI codes. Not sure what the best way to accomplish that is.

**Comment by webbedspace** [comment]
OK, I added a minimal config.

**Comment by rgwood** [comment]
Nice!

The `no_minimal_config` name is bothering me, perhaps more than it should. When reading the code I found myself spending too much time thinking about negation ("OK, it's false by default... and so that means no _no_ minimal config, which means we do use the minimal config..."). Would it make sense for that option to be `minimal_config` with a default of `true` instead?

**Comment by webbedspace** [comment]
Understandable.

File: crates/nu-command/src/viewers/table.rs [hunk 1]
```diff
@@ -351,7 +351,22 @@ fn handle_table_command(
...
+        PipelineData::Value(v, ..) => {
+            // into_string() is used for serialising primitives in PipelineData::write_all_and_flush(),
+            // so the same is used here.
+            let str_representation = v.into_string("", config);
+
+            Ok(Value::String {
+                val: StyleComputer::from_config(engine_state, stack)
+                    .style_primitive(&v)
+                    .color_style
+                    .map(|e| e.paint(str_representation.clone()).to_string())
```
Comment (rgwood) [inline crates/nu-command/src/viewers/table.rs:363]:
```suggestion
                    .map(|e| e.paint(&str_representation).to_string())
```

I believe you can avoid a clone here by passing a reference to `str_representation`.

**Review by rgwood — APPROVED** [review]
LGTM, one comment. Thanks!

**Comment by hustcer** [comment]
It's weird that after this commit I got an 
```console
Error: nu::parser::unexpected_eof (link)

  × Unexpected end of code.
```
error. I have just use `git bisect` to find that this PR may cause the above error.

**Comment by rgwood** [comment]
Bad news: this seems to be adding ANSI escapes to strings passed to externals. That might be the cause of the bug @hustcer is seeing.

Before this PR, `"foo" | clip.exe` copies `foo` to the clipboard.

After this PR, `"foo" | clip.exe` copies `[37mfoo[0m` to the clipboard.

Given that the next release is 1-2 days away (Jan 10), I think we need to find a fix quickly or temporarily revert this PR.

**Comment by fdncred** [comment]
I can think of 2 pretty quick fixes. 

1. Revert the PR
2. Strip the ansi in run-external. I think, not positive, that's where everything gets passed to externals.

**Comment by rgwood** [comment]
I vote to revert given that we don't have much time left to test. I can submit a PR to revert, and then we can hopefully bring back the feature early in the next development cycle.

**Comment by webbedspace** [comment]
So, wait, does this mean that `ls | clip.exe` actually always copied the………
…
image
What the hekate heck???

**Pull request merged**